### PR TITLE
[SW-764] Scroll animation looks weird with shorter names

### PR DESCRIPTION
**Pull request by konradodo**

### CODE REVIEW

File: src/popup/components/Truncate.vue [hunk 1]
```diff
@@ -50,31 +52,38 @@ export default defineComponent({
...
 
     const animationTranslate = ref<string>();
     const animationDuration = ref<string>();
 
+    const calculateTruncate = async () => {
+      shouldScroll.value = false;
+      await nextTick();
+      if (!props.fixed && scroll?.value && container.value?.clientWidth) {
+        const { clientWidth } = container?.value;
+        const { clientWidth: scrollWidth } = scroll?.value;
```
Comment (pstachel) [inline src/popup/components/Truncate.vue:67]: Why do you check the presence of `container` and `scroll` here despite doing that in the `if` clause?

File: src/popup/components/Truncate.vue [hunk 2]
```diff
@@ -50,31 +52,38 @@ export default defineComponent({
...
+      ? props.str.replace(AENS_DOMAIN, '')
+      : props.str);
 
     const animationTranslate = ref<string>();
     const animationDuration = ref<string>();
 
+    const calculateTruncate = async () => {
+      shouldScroll.value = false;
+      await nextTick();
+      if (!props.fixed && scroll?.value && container.value?.clientWidth) {
```
Comment (pstachel) [inline src/popup/components/Truncate.vue:65]:
```suggestion
      if (!props.fixed && scroll.value && container.value?.clientWidth) {
```

File: src/popup/components/Truncate.vue [hunk 3]
```diff
@@ -50,31 +52,38 @@ export default defineComponent({
...
-            if (scrollWidth > offsetWidth) {
-              animationTranslate.value = `-${(animationWidth - (props.fixed ? 0 : 4))}px`;
-              animationDuration.value = `${animationWidth * 100}ms`;
-              shouldScroll.value = true;
-            }
-          }
-        },
-        { immediate: true },
-      );
+      // sometimes setTimeout needed for scroll to calculate proper width
```
Comment (pstachel) [inline src/popup/components/Truncate.vue:83]:
```suggestion
      // sometimes setTimeout is needed for scroll to calculate proper width
```

**Review by pstachel — APPROVED** [review]

File: src/popup/components/TransactionItem.vue [hunk 4]
```diff
@@ -224,6 +226,7 @@ export default defineComponent({
 
       width: 100%;
       color: rgba(variables.$color-white, 0.75);
+      gap: 3px;
```
Comment (CedrikNikita) [inline src/popup/components/TransactionItem.vue:229]: What is the purpose of this property?

Comment (konradodo) [reply src/popup/components/TransactionItem.vue:229]: it's gap between userName and date

File: src/popup/components/Truncate.vue [hunk 5]
```diff
@@ -91,9 +100,12 @@ export default defineComponent({
 </script>
 
 <style lang="scss" scoped>
+
```
Comment (CedrikNikita) [inline src/popup/components/Truncate.vue:103]:
```suggestion
```

File: src/popup/components/Truncate.vue [hunk 6]
```diff
@@ -50,31 +52,38 @@ export default defineComponent({
...
+        if (scrollWidth > clientWidth) {
+          animationTranslate.value = `-${animationWidth}px`;
+          animationDuration.value = `${(animationWidth + 20) * 100}ms`;
+          shouldScroll.value = true;
+        }
+      }
+    };
+
+    watch(
+      () => nameComponent.value,
```
Comment (peronczyk) [inline src/popup/components/Truncate.vue:78]: No need to use callback. Just `watch(nameComponent, ...`.

File: src/popup/components/Truncate.vue [hunk 7]
```diff
@@ -108,8 +118,22 @@ export default defineComponent({
...
+
+      &.ready {
+        opacity: 1;
+      }
+    }
+
     &.scrollable {
+      padding-right: 5px;
       border-radius: 2px;
+
```
Comment (CedrikNikita) [inline src/popup/components/Truncate.vue:136]:
```suggestion
```

File: src/popup/components/Truncate.vue [hunk 8]
```diff
@@ -126,7 +143,6 @@ export default defineComponent({
         animation-duration: var(--animationDuration);
         animation-delay: 1s;
         animation-iteration-count: infinite;
-        animation-timing-function: cubic-bezier(0.42, 0, 0.58, 1);
```
[the remark this answers is not in the document]
Comment (konradodo) [reply src/popup/components/Truncate.vue:129]: animation now looks like this should look

**Pull request merged**